dev/benchmarks/microbenchmarks/lib/foundation/change_notifier_bench.dart

```diff
@@ -6,111 +6,190 @@ import 'package:flutter/foundation.dart';
-      value: removeListenerElapsed * _scale,
-      unit: 'ns per iteration',
-      name: 'removeListener${listenersCount}_iteration',
-    );
-
-    printer.addResult(
-      description: 'notifyListener ($listenersCount listeners)',
-      value: notifyListener * _scale,
-      unit: 'ns per iteration',
-      name: 'notifyListener${listenersCount}_iteration',
-    );
+  void runRemoveListenerWhileNotifyingBenchmark(int iteration,
+      {bool addResult = true}) {
+    const String name = 'removeListenerWhileNotifying';
+
+    final List<VoidCallback> listeners = <VoidCallback>[
+      () {},
+      () {},
+      () {},
+      () {},
+      () {},
+    ];
+    for (int listenerCount = 1; listenerCount <= 5; listenerCount += 1) {
+      final List<_Notifier> notifiers = List<_Notifier>.generate(
+        iteration,
+        (_) {
+          final _Notifier notifier = _Notifier();
+          notifier.addListener(() {
+            // This listener will remove all other listeners. So that only this
+            // one is called and measured.
+            for (int l = 0; l < listenerCount; l += 1) {
+              notifier.removeListener(listeners[l]);
+            }
+          });
+          for (int l = 0; l < listenerCount; l += 1) {
+            notifier.addListener(listeners[l]);
+          }
+          return notifier;
+        },
+        growable: false,
+      );
+
+      final Stopwatch watch = Stopwatch();
+      watch.start();
+      for (int i = 0; i < iteration; i += 1) {
+        notifiers[i].notify();
+      }
+      watch.stop();
+      final int elapsed = watch.elapsedMicroseconds;
+      final double averagePerIteration = elapsed / iteration;
+      if (addResult)
+        printer.addResult(
+          description: '$name ($listenerCount listeners)',
+          value: _makeToStringFriendly(averagePerIteration * _kScale),
+          unit: 'ns per iteration',
+          name: '$name${listenerCount}_iteration',
+        );
+    }
   }
 
+  runAddListenerBenchmark(_kNumWarmUp, addResult: false);
+  runAddListenerBenchmark(_kNumIterations, addResult: true);
+
+  runNotifyListenerBenchmark(_kNumWarmUp, addResult: false);
+  runNotifyListenerBenchmark(_kNumIterations, addResult: true);
+
+  runRemoveListenerBenchmark(_kNumWarmUp, addResult: false);
+  runRemoveListenerBenchmark(_kNumIterations, addResult: true);
+
+  runRemoveListenerWhileNotifyingBenchmark(_kNumWarmUp, addResult: false);
+  runRemoveListenerWhileNotifyingBenchmark(_kNumIterations, addResult: true);
+
   printer.printToStdout();
 }
 
 class _Notifier extends ChangeNotifier {
   void notify() => notifyListeners();
 }
+
+/// round the given double to but 2 decimal places.
+/// this is a workaround to ensure that the printed JSON does not get too long
+/// which can lead to the JSON being cut off in the middle
+/// see issue https://github.com/flutter/flutter/issues/74448
+double _makeToStringFriendly(double d) {
+  return double.parse(d.toStringAsFixed(2));
+}
```

Comment:
This is really hacky. Let's avoid that. I am hoping that just shortening the names should be enough.

Reply:
yeah, I agree. 
Is there a way to run the Mac_android microbenchmarks manually though to see if they would fail? They don't run on normal PR's, right?